src/test/qrcode.spec.js

```diff
@@ -0,0 +1,28 @@
+import { Selector } from "testcafe";
+
+fixture("Token Document Rendering").page`http://localhost:3000`;
+
+const Document = "./fixture/ebl-with-qrcode.json";
+const IframeBlock = Selector("#iframe");
+const SampleTemplate = Selector("#root");
+const DocumentStatus = Selector("#document-status");
+const IssuedByDomainName = Selector("#issuedby .domain");
+
+const validateTextContent = async (t, component, texts) =>
+  texts.reduce(async (_prev, curr) => t.expect(component.textContent).contains(curr), Promise.resolve());
+
+test("UI renders QR code correctly when present in the document", async (t) => {
+  const container = Selector("#certificate-dropzone");
+  await container();
+  await t.setFilesToUpload("input[type=file]", [Document]);
+
+  await DocumentStatus.with({ visibilityCheck: true })();
+
+  const qrcodeButtonElement = await Selector("button").withAttribute("aria-label", "document-utility-qr-button");
+  await t.click(qrcodeButtonElement);
+  await validateTextContent(t, IssuedByDomainName, ["TRADETRUST.IO"]);
+
+  await t.switchToIframe(IframeBlock);
+
+  await validateTextContent(t, SampleTemplate, ["BILL OF LADING FOR OCEAN TRANSPORT OR MULTIMODAL TRANSPORT"]);
```

Comment:
i think after click on qrCode button, maybe check if the qrCode div exists? else this test doesn't seem to check anything other than text rendered correctly from ebl (which is not the case we wanna test?)

Reply:
actually it does, qr button wouldn't even exists in the first place to be clicked 🤔. maybe can ignore my comment